python/composio/cli/workspaces.py

```diff
@@ -0,0 +1,138 @@
+"""
+Composio workspace manager.
+"""
+
+import typing as t
+from datetime import datetime
+from uuid import uuid4
+
+import click
+
+from composio.cli.context import Context, pass_context
+from composio.cli.utils.decorators import handle_exceptions
+from composio.cli.utils.params import EnumParam
+from composio.client.collections import ComposioWorkspaceStatus
+from composio.client.exceptions import HTTPError
+
+
+@click.group(name="workspaces", invoke_without_command=True)
+@click.option(
+    "--status",
+    type=EnumParam(
+        cls=ComposioWorkspaceStatus,
+    ),
+    help="Workspace status filter.",
+    required=False,
+)
+@handle_exceptions()
+@pass_context
+def _workspaces(
+    context: Context,
+    status: t.Optional[ComposioWorkspaceStatus] = None,
+) -> None:
+    """List workspaces available for your account"""
+    if context.click_ctx.invoked_subcommand:
+        return
```

Comment:
In the `_get_workspace_id` function, there's a potential issue where the error message might display `None` for the ID in the exception message. This happens because the variable `id` is checked for `None` after the exception message is constructed. Consider revising the error handling to ensure the message is meaningful and accurate.